src/components/AddApplication/AddApplicationDescription.js

```diff
@@ -15,9 +15,14 @@ import RemoveAppModal from './RemoveAppModal';
 import ApplicationList from '../ApplicationsList/ApplicationList';
 import RedirectNoId from '../RedirectNoId/RedirectNoId';
 
-const AddApplicationDescription = ({ source, sourceTypes }) => {
+const AddApplicationDescription = ({ match: { params: { id } } }) => {
     const [removingApp, setApplicationToRemove] = useState({});
 
+    const sourceTypes = useSelector(({ providers }) => providers.sourceTypes);
+    const entities = useSelector(({ providers }) => providers.entities);
+
+    const source = entities.find(source => source.id  === id);
```

Comment:
This can be done directly in the selector so you might not need the entities subscription